pkg/manifests/manifests.go

```diff
@@ -0,0 +1,169 @@
+package manifests
+
+import (
+	"bytes"
+	"io"
+
+	ingressv1alpha1 "github.com/openshift/cluster-ingress-operator/pkg/apis/ingress/v1alpha1"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/yaml"
+)
+
+const (
+	RouterNamespace          = "assets/router/namespace.yaml"
+	RouterServiceAccount     = "assets/router/service-account.yaml"
+	RouterClusterRole        = "assets/router/cluster-role.yaml"
+	RouterClusterRoleBinding = "assets/router/cluster-role-binding.yaml"
+	RouterDaemonSet          = "assets/router/daemonset.yaml"
+	RouterServiceCloud       = "assets/router/service-cloud.yaml"
+)
+
+func MustAssetReader(asset string) io.Reader {
+	return bytes.NewReader(MustAsset(asset))
+}
+
+// Factory knows how to create ingress-related cluster resources from manifest
+// files. It provides a point of control to mutate the static resources with
+// provided configuration.
+type Factory struct {
+}
+
+func NewFactory() *Factory {
+	return &Factory{}
+}
+
+func (f *Factory) RouterNamespace() (*corev1.Namespace, error) {
+	ns, err := NewNamespace(MustAssetReader(RouterNamespace))
+	if err != nil {
+		return nil, err
+	}
+	return ns, nil
+}
+
+func (f *Factory) RouterServiceAccount() (*corev1.ServiceAccount, error) {
+	sa, err := NewServiceAccount(MustAssetReader(RouterServiceAccount))
+	if err != nil {
+		return nil, err
+	}
+	return sa, nil
+}
+
+func (f *Factory) RouterClusterRole() (*rbacv1.ClusterRole, error) {
+	cr, err := NewClusterRole(MustAssetReader(RouterClusterRole))
+	if err != nil {
+		return nil, err
+	}
+	return cr, nil
+}
+
+func (f *Factory) RouterClusterRoleBinding() (*rbacv1.ClusterRoleBinding, error) {
+	crb, err := NewClusterRoleBinding(MustAssetReader(RouterClusterRoleBinding))
+	if err != nil {
+		return nil, err
+	}
+	return crb, nil
+}
+
+func (f *Factory) RouterDaemonSet(cr *ingressv1alpha1.ClusterIngress) (*appsv1.DaemonSet, error) {
+	ds, err := NewDaemonSet(MustAssetReader(RouterDaemonSet))
+	if err != nil {
+		return nil, err
+	}
+
+	name := "router-" + cr.Name
+
+	ds.Name = name
+
+	ds.Labels = map[string]string{
+		"app": "router",
+	}
+
+	ds.Spec.Selector = &metav1.LabelSelector{
+		MatchLabels: map[string]string{
+			"app": name,
+		},
+	}
+
+	ds.Spec.Template.Labels["app"] = name
+
+	return ds, nil
+}
+
+func (f *Factory) RouterServiceCloud(cr *ingressv1alpha1.ClusterIngress) (*corev1.Service, error) {
+	s, err := NewService(MustAssetReader(RouterServiceCloud))
+	if err != nil {
+		return nil, err
+	}
+
+	name := "router-" + cr.Name
+
+	s.Name = name
+
+	s.Labels = map[string]string{
+		"app": "router",
```

Comment:
Should this be `"app": name`? Maybe it doesn't really matter for the service's label, as opposed to its selector, but it looks a little strange, like possibly an oversight.

Reply:
Similar discussion with the daemonset... might make sense to add a label to each that lets us establish they're "owned by" a given ClusterIngress via the name since this namespace could have multiple sets of resources per ClusterIngress.

Worth doing in a followup.